public/service-worker.js

```diff
@@ -0,0 +1,40 @@
+// https://deanhume.com/create-a-really-really-simple-offline-page-using-service-workers/
+'use strict';
+
+var cacheVersion = 1;
+var currentCache = {
+  offline: 'govuk-calendar-cache-' + cacheVersion
+};
+const offlineUrls = [
+  '/bank-holidays',
+  '/when-do-the-clocks-change'
+];
+
+this.addEventListener('install', event => {
+  this.skipWaiting();
+  event.waitUntil(
+    caches.open(currentCache.offline).then(function(cache) {
+      return cache.addAll(offlineUrls);
+    })
+  );
+});
+
+this.addEventListener('fetch', event => {
+  // request.mode = navigate isn't supported in all browsers
+  // so include a check for Accept: text/html header.
+  if (event.request.mode === 'navigate' || (event.request.method === 'GET' && event.request.headers.get('accept').includes('text/html'))) {
+    event.respondWith(
+      fetch(event.request.url).catch(error => {
+        // Return the offline page
```

Comment:
This reads like we're serving a "You are offline, please hit 'back'" kind of page, when really we're serving the copy of the page the user has visited in the past, if I've understood this...? 😄 

```suggestion
        // Return the cached page if it exists
```